File: 100 days of leetcode/Day4.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        pri = {
            'I' : 1,
            'V' : 5,
            'X' : 10,
            'L' : 50,
            'C' : 100,
            'D' : 500,
            'M' : 1000
            }
        sum = 0
        prev = pri[s[0]]
        sum = sum + prev
        for i in range(1, len(s)):
            if prev < pri[s[i]]:
                sum = sum - prev - prev + pri[s[i]]
                prev = pri[s[i]]
            else:
                sum = sum + pri[s[i]]
                prev = pri[s[i]]
        return sum
        

File: 100 days of leetcode/Day4.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> romanToIntMap = new HashMap<>();
        romanToIntMap.put('I', 1);
        romanToIntMap.put('V', 5);
        romanToIntMap.put('X', 10);
        romanToIntMap.put('L', 50);
        romanToIntMap.put('C', 100);
        romanToIntMap.put('D', 500);
        romanToIntMap.put('M', 1000);

        int sum = 0;
        int prevValue = 0;

        for (int i = s.length() - 1; i >= 0; i--) {
            int currentValue = romanToIntMap.get(s.charAt(i));

            if (currentValue >= prevValue) {
                sum += currentValue;
            } else {
                sum -= currentValue;
            }

            prevValue = currentValue;
        }

        return sum;
    };

}